File: tests/fixtures/ruby_library/library_v1/lib/mylib.rb
# MyLib v1.0.0 - A sample Ruby library for demonstrating API upgrades.

module MyLib
  # A user in the system.
  class User
    attr_reader :id, :name

    def initialize(id, name)
      @id = id
      @name = name
    end

    def to_s
      "User{id=#{@id}, name='#{@name}'}"
    end
  end

  # Connection status.
  module Status
    CONNECTED = :connected
    DISCONNECTED = :disconnected
    ERROR = :error
  end

  # Configuration for the library.
  class Config
    attr_reader :host

    def initialize(host)
      @host = host
    end
  end

  # Database query result item.
  class Item
    attr_reader :key, :value

    def initialize(key, value)
      @key = key
      @value = value
    end
  end

  class << self
    # Get a user by their ID.
    def get_user(id)
      return nil if id <= 0
      User.new(id, "User#{id}")
    end

    # Process raw data bytes.
    def process_data(data)
      data.bytes.map { |b| (b + 1) % 256 }
    end

    # Connect to a host.
    def connect(host)
      raise ArgumentError, "empty host" if host.nil? || host.empty?
      Status::CONNECTED
    end

    # Save data with optional sync.
    def save(data, sync)
      # Force sync if requested
      puts "Saved: #{data}"
    end

    # Query with multiple parameters.
    def query(a, b, c)
      results = []
      results << Item.new(a, b.to_s) if c
      results
    end

    # Find an item by key.
    def find(key)
      Item.new(key, "found")
    end

    # Parse a string.
    def parse(input)
      raise ArgumentError, "empty input" if input.nil? || input.empty?
      input.upcase
    end

    # @deprecated Use fetch_user instead
    def deprecated_fn
      warn "This function is deprecated"
    end
  end

  # Utility functions.
  module Utils
    class << self
      # A helper function that returns a greeting.
      def helper
        "Hello from utils"
      end

      # Format a user for display.
      def format_user(user)
        "User(id=#{user.id}, name=#{user.name})"
      end

      # Create a config map.
      def create_config_map
        { "version" => "1.0.0" }
      end
    end
  end
end

File: tests/fixtures/java_library/library_v1/src/main/java/com/example/MyLib.java
package com.example;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;

/**
 * MyLib v1.0.0 - A sample Java library for demonstrating API upgrades.
 */
public class MyLib {

    /** A user in the system. */
    public static class User {
        private final long id;
        private final String name;

        public User(long id, String name) {
            this.id = id;
            this.name = name;
        }

        public long getId() { return id; }
        public String getName() { return name; }

        @Override
        public String toString() {
            return "User{id=" + id + ", name='" + name + "'}";
        }
    }

    /** Connection status. */
    public enum Status {
        CONNECTED,
        DISCONNECTED,
        ERROR
    }

    /** Configuration for the library. */
    public static class Config {
        private final String host;

        public Config(String host) {
            this.host = host;
        }

        public String getHost() { return host; }
    }

    /** Database query result item. */
    public static class Item {
        private final String key;
        private final String value;

        public Item(String key, String value) {
            this.key = key;
            this.value = value;
        }

        public String getKey() { return key; }
        public String getValue() { return value; }
    }

    /** Get a user by their ID. */
    public static Optional<User> getUser(long id) {
        if (id > 0) {
            return Optional.of(new User(id, "User" + id));
        }
        return Optional.empty();
    }

    /** Process raw data bytes. */
    public static byte[] processData(byte[] data) {
        byte[] result = new byte[data.length];
        for (int i = 0; i < data.length; i++) {
            result[i] = (byte) (data[i] + 1);
        }
        return result;
    }

    /** Connect to a host. */
    public static Status connect(String host) throws Exception {
        if (host == null || host.isEmpty()) {
            throw new Exception("empty host");
        }
        return Status.CONNECTED;
    }

    /** Save data with optional sync. */
    public static void save(String data, boolean sync) throws Exception {
        if (sync) {
            // Force sync
        }
        System.out.println("Saved: " + data);
    }

    /** Query with multiple parameters. */
    public static List<Item> query(String a, int b, boolean c) {
        List<Item> results = new ArrayList<>();
        if (c) {
            results.add(new Item(a, String.valueOf(b)));
        }
        return results;
    }

    /** Find an item by key. */
    public static Item find(String key) {
        return new Item(key, "found");
    }

    /** Parse a string. */
    public static String parse(String input) throws Exception {
        if (input == null || input.isEmpty()) {
            throw new Exception("empty input");
        }
        return input.toUpperCase();
    }

    /** @deprecated Use fetchUser instead */
    @Deprecated
    public static void deprecatedFn() {
        System.out.println("This function is deprecated");
    }

    /** Utility functions. */
    public static class Utils {
        /** A helper function that returns a greeting. */
        public static String helper() {
            return "Hello from utils";
        }

        /** Format a user for display. */
        public static String formatUser(User user) {
            return "User(id=" + user.getId() + ", name=" + user.getName() + ")";
        }

        /** Create a config map. */
        public static Map<String, String> createConfigMap() {
            Map<String, String> map = new HashMap<>();
            map.put("version", "1.0.0");
            return map;
        }
    }
}
